SYCL/ESIMD/api/functional/ctors/ctor_copy.cpp

```diff
@@ -0,0 +1,153 @@
+// method that let call simd constructor for const reference test case
+struct const_ref {
+  static std::string get_description() { return "const reference"; }
+
+  template <typename DataT, int NumElems>
+  static simd<DataT, NumElems>
+  call_simd_by_const_ref(const simd<DataT, NumElems> &simd_by_const_ref) {
+    return simd_by_const_ref;
+  }
+
+  template <typename DataT, int NumElems>
+  static simd<DataT, NumElems> call_simd_ctor(const DataT *ref_data) {
+    simd<DataT, NumElems> source_simd{};
+    source_simd.copy_from(ref_data);
+    return call_simd_by_const_ref<DataT, NumElems>(
+        simd<DataT, NumElems>(source_simd));
+  }
+};
+
+// The main test routine.
+// Use struct with overloaded call operator to have able for iterating over
+// pre-determined datatypes.
+template <typename DataT, int NumElems, typename TestCaseT> struct test {
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    bool passed{true};
+
+    vector_with_allocator<DataT> ref_data{allocator_type<DataT>{queue}};
+    generate_ref_data<DataT, NumElems>(ref_data);
+
+    auto result_data = call_simd<DataT, NumElems, TestCaseT>(queue, ref_data);
+
+    for (size_t it = 0; it < ref_data.size(); it++) {
+      if (ref_data[it] !=
+          result_data[it]) { // here should be bitwise comparison
+        passed = false;
+        log::fail<NumElems>(
+            "Simd by " + TestCaseT::get_description() +
+                " failed, retrieved: " + std::to_string(result_data[it]) +
+                ", expected: " + std::to_string(ref_data[it]),
+            data_type);
+      }
+    }
+
+    return passed;
+  }
+};
+
+template <typename DataT, typename TestT>
+using run_test_with_one_elem = test<DataT, 1, TestT>;
+
+template <typename DataT, typename TestT>
+using run_test_with_eight_elems = test<DataT, 8, TestT>;
+
+template <typename DataT, typename TestT>
+using run_test_with_sixteen_elems = test<DataT, 16, TestT>;
+
+template <typename DataT, typename TestT>
+using run_test_with_thirty_two_elems = test<DataT, 32, TestT>;
+
+template <typename TestT, typename... T>
+bool run_verification_with_chosen_test_type(
+    sycl::queue &queue, const named_type_pack<T...> &types) {
+  bool passed{true};
+
+  passed &= for_all_types<run_test_with_one_elem, TestT>(types, queue);
+  passed &= for_all_types<run_test_with_eight_elems, TestT>(types, queue);
+  passed &= for_all_types<run_test_with_sixteen_elems, TestT>(types, queue);
+  passed &= for_all_types<run_test_with_thirty_two_elems, TestT>(types, queue);
+  return passed;
+}
+
+int main(int argc, char **argv) {
+  sycl::queue queue{esimd_test::ESIMDSelector{},
+                    esimd_test::createExceptionHandler()};
+
+  bool passed{true};
+
+  auto types{get_tested_types<tested_types::all>()};
+
+  passed &= run_verification_with_chosen_test_type<initializer>(queue, types);
+  passed &=
+      run_verification_with_chosen_test_type<var_declaration>(queue, types);
```

Comment:
```suggestion
  ok &= run_verification_with_chosen_test_type<var_declaration>(queue, types);
```
or
```suggestion
  passed &= run_verification_for_type<var_declaration>(queue, types);
```
As a minor suggestion: by shorter names we are able to fit in a single line, so that `var_declaration` parameter is clearly on the same indent level with any other line

Reply:
I've done it [here](https://github.com/intel/llvm-test-suite/pull/578/commits/ec90c5248212600055572f138d681b54f6ab78c0#diff-e20a89556dfa9ec98f8eec4bea698de3975b780ff24a3bde1ced890af1d16581R143-R146)